src/t_string.c

```diff
@@ -38,7 +38,7 @@ int getGenericCommand(client *c);
  *----------------------------------------------------------------------------*/
 
 static int checkStringLength(client *c, long long size) {
-    if (!(c->flags & CLIENT_MASTER) && size > server.proto_max_bulk_len) {
+    if (!mustObeyClient(c) && size > server.proto_max_bulk_len) {
```

Comment:
The LCS command also does a maxmemory check.

Reply:
not sure what you mean...
LCS doesn't unlike SETRANGE, SETBIT, etc, LCS doesn't create a string object.
it does return a huge string, but for that we have output buffer limits (not max_bulk_len).
p.s. it also uses trymalloc and fails gracefully instead of panic.

also, for the purpose of this PR, it's not propagated to replicas or AOF.